File: tests/.hooks/php/primary/src/Hooks/CustomSecurityHook.php
<?php

declare(strict_types=1);

namespace OpenAPI\OpenAPI\Hooks;

use OpenAPI\OpenAPI\Models\Shared\Security;
use Psr\Http\Message\RequestInterface;

class CustomSecurityHook implements BeforeRequestHook
{
    public function beforeRequest(BeforeRequestContext $context, RequestInterface $request): RequestInterface
    {
        $request = $request->withHeader('Idempotency-Key', 'some-key');

        switch ($context->operationID) {
            case 'customSchemeAppId':
                if ($context->securitySource === null) {
                    throw new \InvalidArgumentException('Security source is null');
                }

                $security = $context->securitySource->call($context);

                if (! $security instanceof Security) {
                    throw new \InvalidArgumentException('Security source is not of type Security');
                }

                if ($security->customSchemeAppId === null) {
                    throw new \RuntimeException('CustomSchemeAppID security is not defined');
                }

                $request = $request->withAddedHeader('X-Security-App-Id', $security->customSchemeAppId->appId);
                $request = $request->withAddedHeader('X-Security-Secret', $security->customSchemeAppId->secret);
                break;
        }

        return $request;
    }
}

File: tests/.hooks/ruby/primary/lib/open_api_sdk/sdk_hooks/custom_security_hook.rb
# typed: true
# frozen_string_literal: true

require_relative "./types"

module OpenApiSDK
  module SDKHooks
    class CustomSecurityHook < AbstractSDKHook
      extend T::Sig

      sig do
        override
          .params(
            hook_ctx: BeforeRequestHookContext,
            request: Faraday::Request
          )
          .returns(Faraday::Request)
      end
      def before_request(hook_ctx:, request:)
        request.headers["Idempotency-Key"] = "some-key"

        case hook_ctx.operation_id
        when "customSchemeAppId"

          raise ArgumentError, "Security source is null" if hook_ctx.security_source.nil?

          security = T.must(hook_ctx.security_source).call
          raise ArgumentError, "Security source is not of type Security" unless security.is_a?(Models::Shared::Security)

          raise "CustomSchemeAppID security is not defined" if security.custom_scheme_app_id.nil?

          request.headers["X-Security-App-Id"] = security.custom_scheme_app_id.app_id
          request.headers["X-Security-Secret"] = security.custom_scheme_app_id.secret
        end

        request
      end

    end
  end
end
